specification/eventgrid/Azure.Messaging.EventGrid/main.tsp

```diff
@@ -161,131 +165,129 @@ namespace Azure.Messaging.EventGrid {
+    >;
+  
   @doc("Publish Batch Cloud Event to namespace topic. In case of success, the server responds with an HTTP 200 status code with an empty JSON object in response. Otherwise, the server can return various error codes. For example, 401: which indicates authorization failure, 403: which indicates quota exceeded or message is too large, 410: which indicates that specific topic is not found, 400: for bad request, and 500: for internal server error. ")  
-  @route("/topics/{topicName}:publish", {shared: true})
-  @post op PublishCloudEvents is Azure.Core.RpcOperation<{
-    @doc("content type")
-    @header("content-type")
-    contentType: "application/cloudevents-batch+json; charset=utf-8";
+  @sharedRoute
+  @action("publish")
+  op publishCloudEvents is standardOperations.ResourceAction<
+    Topic,
+    {
+      @doc("content type")
+      @header("content-type")
+      contentType: "application/cloudevents-batch+json; charset=utf-8";
+
+      #suppress "@azure-tools/typespec-azure-core/request-body-problem" "Not using a container model for CloudEvent is intentional. Went through Stewardship Board review (#21880)."
+      @doc("Array of Cloud Events being published.")
+      @body
+      events: CloudEvent[];
+    }, {@body _: PublishResult}>;
 
-    @doc("Topic Name.")
-    @path
-    topicName: string;
-
-    @doc("Array of Cloud Events being published.")
-    @body
-    events: CloudEvent[];
-  }, PublishResult>;
 
   // Receive Operation:
   // POST https://{namespaceName}.{region}.eventgrid.azure.net/topics/{topicName}/eventsubscriptions/{eventSubscriptionName}:receive?api-Version={apiVersion}&maxWaitTime=60&maxEvents={maxEvents}
    
   @doc("Receive Batch of Cloud Events from the Event Subscription.")
-  @route("/topics/{topicName}/eventsubscriptions/{eventSubscriptionName}:receive")
-  @post op ReceiveCloudEvents is Azure.Core.RpcOperation<{
-    @doc("Topic Name.")
-    @path
-    topicName: string;
-
-    @doc("Event Subscription Name.")
-    @path
-    eventSubscriptionName: string;
-
-    @doc("Max Events count to be received. Minimum value is 1, while maximum value is 100 events. If not specified, the default value is 1.")
-    @query
-    maxEvents?: int32 = 1;
-
-    @doc("Max wait time value for receive operation in Seconds. It is the time in seconds that the server approximately waits for the availability of an event and responds to the request. If an event is available, the broker responds immediately to the client. Minimum value is 10 seconds, while maximum value is 120 seconds. If not specified, the default value is 60 seconds.")
-    @encode("seconds", int32)
-    @query
-    maxWaitTime?: duration;
+  @action("receive")
+  op receiveCloudEvents is standardOperations.ResourceAction<
+    EventSubscription,
+    {
+      @doc("Max Events count to be received. Minimum value is 1, while maximum value is 100 events. If not specified, the default value is 1.")
+      @query
+      maxEvents?: int32 = 1;
```

Comment:
You can define min and max value
https://microsoft.github.io/typespec/standard-library/built-in-decorators#@maxValue